src/main/java/ch/qos/logback/core/rolling/AuditLogsS3FixedWindowRollingPolicy.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ch.qos.logback.core.rolling;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.nike.cerberus.util.ConfigUtils;
+import com.typesafe.config.Config;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Rolling policy that will copy audit logs to S3 if enabled when the logs roll.
+ */
+public class AuditLogsS3FixedWindowRollingPolicy<E> extends TimeBasedRollingPolicy<E> {
+
+    private ExecutorService executor = Executors.newFixedThreadPool(1);
+    private final String ENV_NAME = System.getProperty("@environment");
+
+    private AmazonS3 amazonS3;
+    private String bucket;
+    private String bucketRegion;
+    private final Config config = ConfigUtils.getAppConfigMergedWithCliGeneratedProperties();
+
+    private AmazonS3 getAmazonS3() {
+        if (amazonS3 == null) {
+            amazonS3 = AmazonS3Client.builder().withRegion(getBucketRegion()).build();
+        }
+        return amazonS3;
+    }
+
+    private String getBucket() {
+        if (bucket == null) {
+            bucket = config.hasPath("cms.audit.bucket") ? config.getString("cms.audit.bucket") : "";
+        }
+        return bucket;
+    }
+
+    private String getBucketRegion() {
+        if (bucketRegion == null) {
+            bucketRegion = config.hasPath("cms.audit.bucket_region") ? config.getString("cms.audit.bucket_region") : "";
+        }
+        return bucketRegion;
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        // add a hook on JVM shutdown
+        Runtime.getRuntime().addShutdownHook(new Thread(new ShutdownHookRunnable()));
+    }
+
+    @Override
+    public void rollover() throws RolloverFailure {
+        super.rollover();
+
+        if (isS3CopyEnabled()) {
+            uploadAndThenDeleteFileToS3Async(timeBasedFileNamingAndTriggeringPolicy.getElapsedPeriodsFileName() + ".gz");
+        }
+    }
+
+    private void uploadAndThenDeleteFileToS3Async(String filename) {
+
+        final File rolledLogFile = new File(filename);
+
+        // if file does not exist or empty, do nothing
+        if (!rolledLogFile.exists() || rolledLogFile.length() == 0) {
+            return;
+        }
+
+        executor.execute(() -> {
+            try {
+                getAmazonS3().putObject(getBucket(), ENV_NAME + "/audit-logs/" + rolledLogFile.getName(), rolledLogFile);
+                rolledLogFile.delete();
+            } catch (Exception ex) {
+                ex.printStackTrace();
```

Comment:
shouldn't this be a logger?

Reply:
Also, do we want to catch here?

Reply:
You don't have access to the logger in side the log back internals

Reply:
If that is the case, how about also adding a System.err with error message?  Also, maybe a comment?